### ISSUE TITLE
Attachments sometimes flicker for a frame

### ISSUE THREAD

**Issue by appgurueu** [minetest/minetest#11991]
##### Minetest version
```
5.5-dev
```

##### Summary
Animated models and attachments sometimes flicker for a frame. This is a "heisenbug" and hard to observe; the attachment issue appears to occur more frequently though. See [this video](https://youtu.be/gDoFXkCg3yY). Watch key moments at 0.25x speed to see the bug.

Attachment bug (diamond axe temporarily rotated by 180° around the player model):
![Image](https://user-images.githubusercontent.com/34514239/151026545-fc9b73c7-ac34-4a05-947d-6dc9367fc3ec.png)

##### Steps to reproduce
MTG + wield3d or visible_wielditem mod, switch to 3rd person view, reduce viewrange, enable fast, run around.

**Comment by SmallJoker** [comment]
sfan5 already mentioned *somewhere* that the model disappears for one frame when crossing a re-positioning border each 200m. Although I cannot find the correct issue #...

**Comment by sfan5** [comment]
duplicate of #10027

**Comment by appgurueu** [comment]
@sfan5 only partially a duplicate; the attachment issue is lacking in #10027. I changed title and description accodingly, please reopen. The attachment issue is in fact more annoying to me.

**Comment by runsy** [comment]
Yes, an issue. For examples:

- In my Petz mod a Parrot is atached to the shoulder of the player model, and when it walks the rotation ghost arises. The parrot flickers.
- My "The Samz" player model shows this issue with wielded items.

**Comment by appgurueu** [comment]
There also is [this recent thread on the forum](https://forum.minetest.net/viewtopic.php?t=26308&p=390567) that seems to deal with it.

**Comment by sfan5** [comment]
Needs retesting.

**Comment by appgurueu** [comment]
I think I can explain what was probably happening here: The negative scale bug leads to wrong joint transforms (the 180° Y rotation). Attachments are off by one frame (#14818), so Sam looks correct, and the attachment looks wrong exactly as one would expect.

It should be gone for certain once #15687 is merged (and not that it should be necessary, but I will touch that code again later to avoid calculating the scale from the transformation matrix at all by instead only calculating the matrix from the animated TRS properties).
